src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2748,6 +2748,23 @@ protected function hasCast($key)
     }
 
     /**
+     * Determine whether a value is Date / DateTime castable for inbound manipulation.
+     *
+     * @param  string  $key
+     * @return bool
+     */
+    protected function isDateCastable($key)
+    {
+        if ($this->hasCast($key)) {
+            return in_array(
+                $this->getCastType($key), ['date', 'datetime'], true
```

Comment:
That way it's super readable because the code is clearly saying if has cast and is in array, return true, rather than some nested stuff that's harder to read and see the intent so quickly.